scipy/stats/tests/test_continuous_basic.py

```diff
@@ -465,28 +463,18 @@ def test_method_of_moments():
     npt.assert_almost_equal(loc+scale, b, decimal=4)
 
 
-def check_sample_meanvar_(distfn, arg, m, v, sm, sv, sn, msg):
+def check_sample_meanvar_(popmean, popvar, sample):
     # this did not work, skipped silently by nose
-    if np.isfinite(m):
-        check_sample_mean(sm, sv, sn, m)
-    if np.isfinite(v):
-        check_sample_var(sv, sn, v)
+    if np.isfinite(popmean):
+        check_sample_mean(sample, popmean)
+    if np.isfinite(popvar):
+        check_sample_var(sample.var(), len(sample), popvar)
 
 
-def check_sample_mean(sm, v, n, popmean):
-    # from stats._stats_py.ttest_1samp(a, popmean):
-    # Calculates the t-obtained for the independent samples T-test on ONE group
-    # of scores a, given a population mean.
-    #
-    # Returns: t-value, two-tailed prob
-    df = n-1
-    svar = ((n-1)*v) / float(df)    # looks redundant
-    t = (sm-popmean) / np.sqrt(svar*(1.0/n))
-    prob = betainc(0.5*df, 0.5, df/(df + t*t))
-
-    # return t,prob
-    npt.assert_(prob > 0.01, 'mean fail, t,prob = %f, %f, m, sm=%f,%f' %
-                (t, prob, popmean, sm))
+def check_sample_mean(sample, popmean):
+    # Checks for unlikely difference between sample mean and population mean
+    prob = stats.ttest_1samp(sample, popmean).pvalue
+    assert prob > 0.01
```

Comment:
If this is not what the test was intended to do, then at least this PR brings that to light, and we can update the test to make it's purpose more clear.

Reply:
@josef-pkt Does this change look OK? This doesn't give exactly the same p-value, but it seems to be what the test was intended to do.

I'd also appreciate your thoughts on whether the top post of this PR resolves gh-2623.